core/shared/src/main/scala/org/http4s/multipart/PartValue.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.multipart
+
+import fs2.Stream
+import fs2.io.file.Files
+import fs2.io.file.Flags
+import fs2.io.file.Path
+import org.http4s.EntityBody
+import org.http4s.Header
+
+/** Generic representation of typical Multipart Part bodies, as either a string or a file path.
+  * Produced by [[PartReceiver.toMixedBuffer]].
+  *
+  * Similar to [[Part]], but with less Header-oriented detail, and a clearer distinction
+  * between in-memory data vs on-disk data.
+  */
+sealed trait PartValue {
+
+  /** Returns a byte-stream representation of the part's body.
+    *
+    * For `OfString` parts, the underlying string is piped through a UTF-8 encoder.
+    * For `OfFile` parts, the underlying file's content will be read from disk.
+    *
+    * @return The part's body as a stream of bytes
+    */
+  def bytes[F[_]: Files]: EntityBody[F]
+
+  /** Fold over the different PartValue subtypes to compute a value
+    *
+    * @param onString Called with the underlying String if this part is an `OfString`
+    * @param onFile Called with the underlying filename and file path if this part is an `OfFile`
+    * @return The value returned by either `onString` or `onFile`
+    */
+  def fold[A](onString: String => A, onFile: (String, Path) => A): A
+
+  /** Builds a new [[Part]] from this `PartValue`.
+    *
+    * @param name The name of the part
+    * @param headers Any extra headers to add to the part. `Content-Disposition` is added
+    *                automatically based on the `name`, and in the case of `OfFile` parts,
+    *                the filename
+    * @return a new [[Part]] based on the given name, applicable filename, and body of this `PartValue`.
+    */
+  def toPart[F[_]: Files](name: String, headers: Header.ToRaw*): Part[F] = fold(
+    s => Part.formData(name, s, headers: _*),
+    (filename, _) => Part.fileData(name, filename, bytes[F], headers: _*),
+  )
+}
+object PartValue {
```

Comment:
It seems strange to me that there's no `OfBytes`.  It has to go to a file to be a binary?

Reply:
Hm.. `PartValue` was intended as the representation of a decoded part from `MultipartReceiver.auto`, which picks between decoding the part to a String or dumping it to a File depending on the part headers. An `OfBytes` case would imply the `auto` would sometimes decide not to decode a regular part to text, and I don't know offhand what kind of edge cases might motivate that. 

If there was a relevant "business logic" motiviation, then the user wouldn't use `MultipartReceiver.auto`; instead they'd set up a custom MultipartReceiver that decides to use `PartReceiver.toMixedBuffer` for arbitrary "bytes" receiving.

You bringing this up made me realize that the docs I wrote on `trait PartValue` are wrongfully referencing `PartReceiver.toMixedBuffer`. Maybe correcting that will be enough?